spec/src/main/asciidoc/configexamples.asciidoc

```diff
@@ -238,17 +238,25 @@ details.endpoint -> endpoint -> book
 details.getLocation() -> old.location -> Berlin
 ```
 
-In the above two types of injection lookup, the configuration properties class should contain a zero-arg constructor. Otherwise, the behaviour is unspecified. 
-When preforming property lookup, `DeploymentException` will be thrown for the following scenarios:
+In the above two types of injection lookup, the configuration properties class should contain a zero-arg constructor. Otherwise, the behaviour is unspecified.
+When performing property lookup, a `DeploymentException` will be thrown for the following scenarios:
 
 1. The property is missing and neither default value nor the property return type is optional.
 Use one of the following to fix the problem.
 ** Define a value for the property
-** Supply a default value when defining the field. 
-** Use `@ConfigProperty` to provide a default value. 
+** Supply a default value when defining the field.
+** Use `@ConfigProperty` to provide a default value.
 ** Use `Optional<T>` or `OptionalInt`, `OptionalDouble`, `OptionalLong` as the type.
 
 2. The property value cannot be converted to the specified type
 
+If any of the property cannot be found and there is neither default value nor property is not optional, `java.util.NoSuchElementException` will be thrown.
+In order to avoid this, you can supply a default value when defining the field. Alternatively, you can use `@ConfigProperty` to provide a default value.
+You can also use `Optional<T>` or OptionalInt, OptionalDouble, OptionalLong as the type. If any property values cannot be converted to the specified type, `java.lang.IllegalArgumentException` will be thrown.
+
+When using programmatic lookup of the configuration properties, the configuration properties class does not need to be annotated with
```

Comment:
The lines between 257 - 260 should be removed as we don't have the spi any more.